### PR TITLE
API-16005 split doc updates (part 2: NOD)

### PR DESCRIPTION
**Pull request by cilestin**
## Description of change
Brings more of our NOD documentation in line with becoming segmented & having shared schemas across the board.

## Original issue(s)
https://vajira.max.gov/browse/API-16005

### CODE REVIEW

File: modules/appeals_api/spec/docs/v2/nod_spec.rb [hunk 1]
```diff
@@ -151,27 +153,71 @@
...
-      description 'Returns the [JSON Schema](https://json-schema.org/) for the `POST /notice_of_disagreement` endpoint.'
-      security [
-        { apikey: [] }
-      ]
-      produces 'application/json'
+  if DocHelpers.wip_doc_enabled?(:segmented_apis, true)
+    path '/schemas/{schema_type}' do
+      get 'Gets the Notice of Disagreement JSON Schema.' do
+        tags 'Notice of Disagreements'
+        description 'Returns the [JSON Schema](https://json-schema.org/) for the `POST /forms/10182` endpoint.'
```
Comment (caseywilliams) [inline modules/appeals_api/spec/docs/v2/nod_spec.rb:160]: I like including the backticks around `POST /forms/10182` - I just noticed that they're missing in the equivalent description for HLR [here](https://github.com/department-of-veterans-affairs/vets-api/pull/10717/files#diff-39f0e6060ca7b1326e569973b31abb9e07537e03616ec2e8f16f2364e74ad288R350) though

**Review by caseywilliams — APPROVED** [review]
LGTM, just noticed something I missed in #10717

File: modules/appeals_api/spec/docs/v2/nod_spec.rb (hunk 1 above)
Comment (cilestin) [reply modules/appeals_api/spec/docs/v2/nod_spec.rb:160]: Good call! I'll add that to this PR as a fixup

File: modules/appeals_api/spec/support/rswag_config.rb [hunk 2]
```diff
@@ -58,6 +58,7 @@ def config
   def schemas
     a = []
     a << generic_schemas('#/components/schemas')
+    a << contestable_issues_schema('#/components/schemas')
```
Comment (caseywilliams) [inline modules/appeals_api/spec/support/rswag_config.rb:61]: ooc, why does this move to the top?

**Review by caseywilliams — COMMENTED** [review]
LGTM in general, just had a question

File: modules/appeals_api/spec/support/rswag_config.rb (hunk 2 above)
Comment (cilestin) [reply modules/appeals_api/spec/support/rswag_config.rb:61]: At the bottom of this method we merge duplicate keys & then sort the keys alphabetically:

```ruby
 a.reduce(&:merge).sort_by { |k, _| k.to_s.downcase }.to_h
```

**Comment by caseywilliams** [comment]
This still LGTM but it looks like some tests failed this time

**Review by caseywilliams — APPROVED** [review]

**Pull request merged**